### PR TITLE
Fix upload consensus

### PR DESCRIPTION
**Pull request by Hitenjain14**
### Changes
-

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by din-mukhammed — APPROVED** [review]
LGTM

File: zboxcore/sdk/allocation.go [hunk 1]
```diff
@@ -1063,19 +1063,19 @@ func (a *Allocation) addAndGenerateDownloadRequest(
...
-	var connectionID string
-	if len(a.downloadRequests) > 0 {
-		connectionID = a.downloadRequests[0].connectionID
-	} else {
-		connectionID = zboxutil.NewConnectionId()
-	}
 	downloadReq, err := a.generateDownloadRequest(
 		fileHandler, remotePath, contentMode, startBlock, endBlock,
-		numBlocks, verifyDownload, status, connectionID, localFilePath)
+		numBlocks, verifyDownload, status, "", localFilePath)
```
Comment (dabasov) [inline zboxcore/sdk/allocation.go:1068]: why do we set connectionID after?

File: zboxcore/sdk/chunked_upload.go [hunk 2]
```diff
@@ -598,50 +598,67 @@ func (su *ChunkedUpload) processUpload(chunkStartIndex, chunkEndIndex int,
...
+		go func(b *ChunkedUploadBlobber, thumbnailChunkData []byte, pos uint64) {
+			defer wg.Done()
+			body, formData, err := su.formBuilder.Build(
+				&su.fileMeta, blobber.progress.Hasher, su.progress.ConnectionID,
+				su.chunkSize, chunkStartIndex, chunkEndIndex, isFinal, encryptedKey,
+				fileShards[pos], thumbnailChunkData)
+			if err != nil {
+				errC := atomic.AddInt32(&errCount, 1)
+				if errC > int32(su.allocationObj.ParityShards-1) { // If atleast data shards + 1 number of blobbers can process the upload, it can be repaired later
+					wgErrors <- err
```
Comment (dabasov) [inline zboxcore/sdk/chunked_upload.go:655]: goroutines can get stuck here, as only one message will be read from wgErrors

**Review by dabasov — CHANGES_REQUESTED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | hotfix/upload-consensus  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  |   |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | hotfix/upload-consensus  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  |   |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Pull request merged**